### PR TITLE
refactored

### PR DESCRIPTION
**Pull request by Mcorupe**
refactored some state/actions & state/reducers. added some starter stuff for search(probably needs to get looked at closely cuz i didnt add reduxthunk. so definetly check that out as im still learning those.

### CODE REVIEW

File: frontend/src/shared-components/header.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+//need to make header a higher order component
+//use Connect: dispatching actions with mapDispatchToProps
+
+class Header extends React.Component {
+  render() {
+    const {login, signup, logout} = this.props;
+    return (
+      <div className="app">
+        <Header title={header} />
+        <Body title={body} />
```
Comment (signalnerve) [inline frontend/src/shared-components/header.js]:
```js
<Body body={body} />
```

File: frontend/src/shared-components/header.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+
+import { LOGIN, SIGNUP, LOGOUT } from "../actions/header-actions";
+
+
+
+const Header = ({title}) => (<header>{title}</header>);
+const Body = ({title}) => (<body>{title}</body>);
```
Comment (signalnerve) [inline frontend/src/shared-components/header.js]:
```js
const Body = ({body}) => (<body>{body}</body>);
```

File: frontend/src/shared-components/header.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+//need to make header a higher order component
+//use Connect: dispatching actions with mapDispatchToProps
+
+class Header extends React.Component {
+  render() {
+    const {login, signup, logout} = this.props;
```
Comment (signalnerve) [inline frontend/src/shared-components/header.js]: Make sure to get `header` and `body` out of props here!

File: frontend/src/state/actions/search-actions.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+// };
+
+
+
+export const search = () => (dispatch, getState) => {
+    
+  
+    const payload = { //the data[] };
+    // data for redux store is a boolean flag
+    dispatch({ type: LOGOUT, payload });
```
Comment (signalnerve) [inline frontend/src/state/actions/search-actions.js]: This should dispatch `SEARCH_VIDEOS`, I think

File: frontend/src/state/actions/search-actions.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+export const SEARCH_VIDEOS = "SEARCH_VIDEOS";
```
Comment (signalnerve) [inline frontend/src/state/actions/search-actions.js]: mentioning this again after review of #41 - I really think all of these files inside of `actions` should be like `actions/search`, not `actions/search-actions`. Up to you but seems like unnecessary duplication IMO - make sure to check your imports if you do rename

File: frontend/src/shared-components/header.js (hunk 2 above)
Comment (Mcorupe) [reply frontend/src/shared-components/header.js]: does this mean that header also needs the same changes? cept using Header

Comment (signalnerve) [reply frontend/src/shared-components/header.js]: Nope! You're using two props, `title`, and `body`, just make sure you pass them to the correct places. `Header` is good as-is

**Comment by signalnerve** [comment]
Make sure to request review from someone else on your team, too!

**Pull request merged**